internal/resources/smbshare.go

```diff
@@ -322,6 +207,189 @@ func (m *SmbShareManager) Update(
 	return Done
 }
 
+func (m *SmbShareManager) updatePVC(
+	ctx context.Context,
+	instance *sambaoperatorv1alpha1.SmbShare) Result {
+	// ---
+	destNamespace := instance.Namespace
+	pvc, created, err := m.getOrCreatePvc(
+		ctx, instance, destNamespace)
+	if err != nil {
+		return Result{err: err}
+	} else if created {
+		m.logger.Info("Created PVC")
+		m.recorder.Eventf(instance,
+			EventNormal,
+			ReasonCreatedPersistentVolumeClaim,
+			"Created PVC %s for SmbShare", pvc.Name)
+		return Requeue
+	}
+	// if name is unset in the YAML, set it here
+	instance.Spec.Storage.Pvc.Name = pvc.Name
+	return Done
+}
+
+func (m *SmbShareManager) updateBackend(
+	ctx context.Context,
+	planner *pln.Planner) Result {
+	// ---
+	var (
+		err      error
+		instance = planner.SmbShare
+	)
+	if instance.Annotations == nil {
+		instance.Annotations = map[string]string{}
+	}
+	if planner.IsClustered() {
+		instance.Annotations[serverBackend] = clusteredBackend
+	} else {
+		instance.Annotations[serverBackend] = standardBackend
+	}
+	m.logger.Info("Setting backend",
+		"SmbShare.Namespace", instance.Namespace,
+		"SmbShare.Name", instance.Name,
+		"SmbShare.UID", instance.UID,
+		"backend", instance.Annotations[serverBackend])
+	err = m.client.Update(ctx, instance)
+	if err != nil {
+		m.logger.Error(
+			err,
+			"Failed to update SmbShare",
+			"SmbShare.Namespace", instance.Namespace,
+			"SmbShare.Name", instance.Name,
+			"SmbShare.UID", instance.UID)
+		return Result{err: err}
+	}
+	return Requeue
+}
+
+func (m *SmbShareManager) validateBackend(
+	planner *pln.Planner) Result {
+	// ---
+	var (
+		err      error
+		instance = planner.SmbShare
+	)
+	// As of today, the system is too immature to try and trivially
+	// reconcile a change in availability mode. We use the previously
+	// recorded backend to check if a change is being made after the
+	// "point of no return".
+	// In the future we should certainly revisit this and support
+	// intelligent methods to handle changes like this.
+	b := instance.Annotations[serverBackend]
+	if planner.IsClustered() && b != clusteredBackend {
+		err = fmt.Errorf(
+			"Can not convert SmbShare to clustered instance."+
+				" Current backend: %s",
+			b)
```

Comment:
Style: I find this line-break confusing.

Reply:
OK. It was like that in the previous version of the code before it was moved (you can see this if you use "ignore whitespace" feature).  If you want I can add a follow up patch or if its not a blocker for this PR, we can leave it and the next time we work on the code we can improve the formatting.